numpy/_core/src/umath/string_buffer.h

```diff
@@ -313,12 +464,43 @@ struct Buffer {
 };
 
 
+template <IMPLEMENTED_FUNCTIONS f, ENCODING enc, typename t>
+struct call_buffer_member_function {
+    t operator()(Buffer<enc> buf) {
+        switch (f) {
+            case IMPLEMENTED_FUNCTIONS::ISALPHA:
+                return codepoint_isalpha<enc>(*buf);
+            case IMPLEMENTED_FUNCTIONS::ISDIGIT:
+                return codepoint_isdigit<enc>(*buf);
+            case IMPLEMENTED_FUNCTIONS::ISSPACE:
+                return codepoint_isspace<enc>(*buf);
+            case IMPLEMENTED_FUNCTIONS::STR_LEN:
+                return (t)buf.num_codepoints();
+            case IMPLEMENTED_FUNCTIONS::ISNUMERIC:
+                return codepoint_isnumeric(*buf);
+            case IMPLEMENTED_FUNCTIONS::ISDECIMAL:
+                return codepoint_isdecimal(*buf);
+        }
+    }
+};
+
 template <ENCODING enc>
 inline Buffer<enc>
 operator+(Buffer<enc> lhs, npy_int64 rhs)
 {
-    lhs += rhs;
-    return lhs;
+    switch (enc) {
+        case ENCODING::ASCII:
```

Comment:
I feel it might be a bit clearer if we adjust `Buffer.after` in `operator+=` and then just call that from here. Am I missing something?

Reply:
If I used `operator+=` I'd mutate `lhs`, which I don't want to do.